### ISSUE TITLE
Join functions

### ISSUE THREAD

**Issue by evinism** [evinism/mistql#7]
Can imagine:

`join.left`
`join.right`
`join.outer`
`join.inner`

**Comment by cfilipov** [comment]
You have a `join` function which is used for concatenating strings with a separator which I feel can be confusing with the relational joins proposed. Maybe the existing `join` should be `concat` or something?

**Comment by evinism** [comment]
yeah, that naming conflict has definitely been a concern of mine for a bit -- i opted for join as the inverse of split, but that might have been a poor long-term choice. I like concat though! 

I also think now is the right time to correct this -- if i don't it's going to be very annoying to correct in the future.
